components/patient-data/migrations/src/main/java/org/phenotips/data/internal/R70190PhenoTips1280DataMigration.java

```diff
@@ -0,0 +1,290 @@
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<EntityReference> entityResolver;
+
+    @Override
+    public String getDescription()
+    {
+        return "Migrate all existing gene values to the GeneClass objects";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(70190);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override
+    public Object doInHibernate(Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        DocumentReference patientClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(PATIENT_CLASS);
+        DocumentReference investigationClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(INVESTIGATION_CLASS);
+        DocumentReference geneClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(GENE_CLASS);
+        DocumentReference rejectedGenesClassReference =
+            R70190PhenoTips1280DataMigration.this.entityResolver.resolve(REJECTED_CLASS);
+
+        Query q =
+            session.createQuery("select distinct o.name from BaseObject o, StringProperty p where o.className = '"
+                + this.serializer.serialize(geneClassReference) + OR
+                + this.serializer.serialize(rejectedGenesClassReference) + OR
+                + this.serializer.serialize(patientClassReference) + "' and p.id.id = o.id and p.id.name = '"
+                + SOLVED_NAME + "' and p.value <> ''");
+
+        @SuppressWarnings("unchecked")
+        List<String> docs = q.list();
+        for (String docName : docs) {
+            XWikiDocument doc =
+                xwiki.getDocument(R70190PhenoTips1280DataMigration.this.resolver.resolve(docName), context);
+            List<String> geneList = new ArrayList<>();
+            setSolvedGenes(doc, xwiki, patientClassReference, geneClassReference, context, session, geneList);
+            setRejectedGenes(doc, xwiki, rejectedGenesClassReference, geneClassReference, context, session, geneList);
+            setCandidateGenes(doc, xwiki, investigationClassReference, geneClassReference, context, session, geneList);
+        }
+
+        return null;
+    }
+
+    private void setSolvedGenes(XWikiDocument doc, XWiki xwiki, DocumentReference patientClassReference,
+        DocumentReference geneClassReference, XWikiContext context, Session session, List<String> geneList)
+        throws HibernateException, XWikiException
+    {
+        BaseObject patient = doc.getXObject(patientClassReference);
+        StringProperty oldTarget = (StringProperty) patient.get(SOLVED_NAME);
+        patient.removeField(SOLVED_NAME);
```

Comment:
@sdumitriu Is it wise to remove a field before creating a new one? Or does it make no difference since the ```save``` call needs to happen before any changes take place?